solutions/usaco-1016.mdx

```diff
@@ -0,0 +1,180 @@
+
+1. If $\texttt{group0}$ is the same as $\texttt{group1}$ (mod 12), Bessie can start from any node.
+2. If $\texttt{group0}$ is 1 smaller than $\texttt{group1}$ (mod 12), Bessie can start from any node in $\texttt{group1}$.
+3. If $\texttt{group1}$ is 1 smaller than $\texttt{group0}$ (mod 12), Bessie can start from any node in $\texttt{group0}$.
+4. If $\texttt{group0}$ and $\texttt{group1}$ differ by at least 2 (mod 12), there's no way to get all clocks to point to 12.
+
+## Video Solution
+
+<Youtube id="vMA54A2AT0I" />
+
+## Implementation
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int N;
+vector<int> edges[100000];
+int A[100000];
+int sum0, sum1, nodes0, nodes1;
+
+void dfs(int i, int color, int par) {
+	// update color/sum
+	if (color == 0) {
+		nodes0++;
+		sum0 += A[i];
+	}
+	else {
+		nodes1++;
+		sum1 += A[i];
+	}
+	
+	for (int j : edges[i]) {				   
+		if (j != par) {
+			// swap colors for the child
+			dfs(j, 1 - color, i);
+		}
+	}
+}
+
+int main() {
+	freopen("clocktree.in", "r", stdin);
+	freopen("clocktree.out", "w", stdout);
+
+	cin >> N;
+
+	for (int i = 0; i < N; i++) {
+		cin >> A[i];
+	}
+
+	for (int i = 1; i < N; i++) {	
+		int a, b;
+		cin >> a >> b;
+		a--, b--;
+		edges[a].push_back(b);
+		edges[b].push_back(a);
+	}
+	
+	dfs(0, 0, -1);
+
+	// same mod means you can start anywhere
+	if ((sum0 % 12) == (sum1 % 12)) {
+		cout << N;
+	}
+	// if group0 is 1 smaller than group1, we must start from group1
+	else if ((sum0 + 1) % 12 == (sum1 % 12)) {
+		cout << nodes1;
+	}
+	// if group1 is 1 smaller than group0, we must start from group0
+	else if ((sum0 % 12) == ((sum1 + 1) % 12)) {
+		cout << nodes0;
+	}
+	// no way to get all clocks pointing to 12
+	else {
+		cout << 0;
+	}
+
+}
+```
+
+</CPPSection>
+
+<JavaSection>
+
+```java
+public class ClockTree {
+	static int[] clocks;
+	static List<ArrayList<Integer>> adj;
+	static int sum0 = 0, sum1 = 0, nodes0 = 0, nodes1 = 0;
+
+	public static void main(String[] args) throws IOException {
+		Kattio io = new Kattio("clocktree");
+		
+		int n = io.nextInt();
+		clocks = new int[n];
+		adj = new ArrayList<>();
+		for (int i = 0; i < n; i++) {
+			clocks[i] = io.nextInt();
+			adj.add(new ArrayList<>());
+		}
+		for (int i = 0; i < n - 1; i++) {
+			int a = io.nextInt() - 1;
+			int b = io.nextInt() - 1;
+			adj.get(a).add(b);
+			adj.get(b).add(a);
+		}
+
+		dfs(0, 0, -1);
+
+		// same mod means you can start anywhere
+		if ((sum0 % 12) == (sum1 % 12)) {
+			io.println(n);;
+		}
+		// if group0 is 1 smaller than group1, we must start from group1
+		else if ((sum0 + 1) % 12 == (sum1 % 12)) {
+			io.println(nodes1);
+		}
+		// if group1 is 1 smaller than group0, we must start from group0
+		else if ((sum0 % 12) == ((sum1 + 1) % 12)) {
+			io.println(nodes0);
+		}
+		// no way to get all clocks pointing to 12
+		else {
+			io.println(0);
+		}
+		io.close();
+	}
+
+	static void dfs(int i, int color, int par) {
```

Comment:
```suggestion
	static void dfs (int i, int color, int par) {
```

Reply:
I think it's convention to not leave a space in between the method name and parameters?
![image](https://user-images.githubusercontent.com/8647920/125383473-4fc0ff00-e34c-11eb-8e3f-4e98d108fe04.png)

Reply:
oh